src/modules/content-tree/content.tree.module.js

```diff
@@ -2,20 +2,79 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import List from './components/list/list.component';
 
+const CLASS_IS_TREE_RESIZING = 'ez-is-tree-resizing';
+
 export default class ContentTreeModule extends Component {
     constructor(props) {
         super(props);
 
+        this.changeContainerWidth = this.changeContainerWidth.bind(this);
+        this.addWidthChangeListener = this.addWidthChangeListener.bind(this);
+        this.removeWidthChangeListener = this.removeWidthChangeListener.bind(this);
+        this._refTreeContainer = React.createRef();
+
         this.state = {
             selectedLocationId: props.selectedLocationId,
             locations: props.preloadedLocations,
+            resizeStartPositionX: 0,
+            containerWidth: 0,
+            resizedContainerWidth: 0,
+            isResizing: false,
         };
     }
 
+    componentWillUnmount() {
+        this.clearDocumentResizingState();
+    }
+
+    changeContainerWidth({ clientX }) {
+        const currentPositionX = clientX;
+
+        this.setState((state) => ({
+            resizedContainerWidth: state.containerWidth + (currentPositionX - state.resizeStartPositionX),
+        }));
+    }
+
+    addWidthChangeListener({ nativeEvent }) {
+        const resizeStartPositionX = nativeEvent.clientX;
+        const containerWidth = parseInt(window.getComputedStyle(this._refTreeContainer.current).width, 10);
+
+        window.document.addEventListener('mousemove', this.changeContainerWidth, false);
+        window.document.addEventListener('mouseup', this.removeWidthChangeListener, false);
+        window.document.body.classList.add(CLASS_IS_TREE_RESIZING);
```

Comment:
I don't see any value in moving this piece of code into a separate method. Moreover, adding event listeners and adding a CSS class to a callback invoked after updating a state doesn't change anything. It's because the listeners are added to `window.document` and `window.document.body` which are not controlled nor modified by React updates.